onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -661,6 +667,12 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
     std::string string_buf;
     model_proto.SerializeToString(&string_buf);
 
+    if (dump_subgraphs_) {
```

Comment:
Thanks. One more place we can dump the model for debugging,
GetSupportedList():
after the line: model_proto.SerializeToString(&string_buf)
Here we can capture onnx-tensorrt parser failures during GetCapability.

Reply:
> Thanks. One more place we can dump the model for debugging,
> GetSupportedList():
> after the line: model_proto.SerializeToString(&string_buf)
> Here we can capture onnx-tensorrt parser failures during GetCapability.

Can I handle that in a separate PR? I have some questions about the GetCapability/GetSupportedList fail path. Also is there potential for duplicate subgraphs getting dumped then.

my main motivation for this PR is to allow dumping of the subgraph corresponding to the fused node referred to by this error message https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L723

Reply:
or maybe a better option would be to directly dump the subgraph when the error is hit and notify the user it was saved. 
this way we don't have to explicitly set some env variable.

Reply:
There are two cases that error could happen in the parser. One is that parser failed when parsing the model. Dumping the model in GetCapability will help this case. The other case is that parser doesn't fail on model parsing, but fail on reporting exceptions to ORT. Then ORT compile (engine creation) will fail. But sure, we can handle the second case in this PR. I will include the first case in next PR.